From b8d9f05e964dd802cbecf1ebbe82ad40c1d1fdba Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Mon, 15 Sep 2014 19:58:47 -0400
Subject: [thermal] intel_powerclamp: add newer cpu ids

Message-id: <20140915195847.40834.19513.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 94802
O-Subject: [PATCH RHEL7.1 BZ1141895] thermal/intel_powerclamp: add newer cpu ids
Bugzilla: 1141895
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1141895

Description:
------------
Add support for Broadwell and Valleyview CPUs

Signed-off-by: Jacob Pan <jacob.jun.pan@linux.intel.com>

RHEL Version Found:
-------------------
RHEL 7.0

Brew:
-----
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7957283

Upstream:
---------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=9a17f56c590cfd0d5b47f3ffa89c297534962e8f

Test Status:
------------
Successfully tested by me using intel-sharkbay-ult-03 which is cpu family 6 model 61.

before the patch the driver doesn't load and the following msg is displayed.
insmod intel_powerclamp.ko
[   95.276432] intel_powerclamp: Intel powerclamp does not run on family 6 model 61
insmod: ERROR: could not insert module intel_powerclamp.ko: No such device

after the patch the driver loads and works.

[root@intel-sharkbay-ult-03 ~]# lsmod
Module                  Size  Used by
sg                     40721  0
intel_powerclamp       18764  0

---------------------------------------------------------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/thermal/intel_powerclamp.c b/drivers/thermal/intel_powerclamp.c
index 00b9372..90f5561 100644
--- a/drivers/thermal/intel_powerclamp.c
+++ b/drivers/thermal/intel_powerclamp.c
@@ -681,8 +681,10 @@ static const struct x86_cpu_id intel_powerclamp_ids[] = {
  { X86_VENDOR_INTEL, 6, 0x2d},
  { X86_VENDOR_INTEL, 6, 0x2e},
  { X86_VENDOR_INTEL, 6, 0x2f},
+ { X86_VENDOR_INTEL, 6, 0x37},
  { X86_VENDOR_INTEL, 6, 0x3a},
  { X86_VENDOR_INTEL, 6, 0x3c},
+ { X86_VENDOR_INTEL, 6, 0x3d},
  { X86_VENDOR_INTEL, 6, 0x3e},
  { X86_VENDOR_INTEL, 6, 0x3f},
  { X86_VENDOR_INTEL, 6, 0x45},
-- 
1.7.1